lms/djangoapps/courseware/views/views.py

```diff
@@ -541,7 +550,7 @@ def register_user_access_warning_messages(request, course_key):
                 )
             )
         else:
-            if not CourseEnrollment.is_enrolled(request.user, course_key):
+            if not CourseEnrollment.is_enrolled(request.user, course_key) and not allow_anonymous:
```

Comment:
Will this line prevent an unenrolled (but logged in) student from enrolling in the class?

Reply:
The idea is that there's no need for enrollment if the course is marked public. Marking a course as public would mean anybody, whether a registered student or not can access the course contents.

Reply:
Yup, I got that idea :) I just wondered if things like grades, etc, would be effected.

Reply:
Ah yes! As per the proposal, the parts of the course that contains exams and grading would not be allowed public access. Do you think having a message like below in the course landing page makes more sense in that case?
![enroll_option](https://user-images.githubusercontent.com/13742492/51544910-ed779980-1e86-11e9-8c00-c6c7ca5da912.png)

Reply:
Fair enough :-)